catalog/core/tests/test_views.py

```diff
@@ -154,41 +177,77 @@ def test_canonical_publication_detail_view(self):
         self.without_login_and_with_login_test(url, after_status=404)
 
     # Test that publication detail is saved successfully or not
-    def test_publication_detail_save_with_all_valid_fields(self):
-        p = models(Publication, container=models(Container),
-                   added_by=models(User),
-                   code_archive_url=st.text(),
-                   url=st.text(), zotero_key=st.just(None)).example()
+    @settings(max_examples=15, perform_health_check=False)
+    @given(GENRATE_PUBLICATION, st.text(min_size=10), st.sampled_from(GET_PLATFORM), st.sampled_from(GET_SPONSORS),
+           st.sampled_from(GET_MODEL_DOCUMENTATION), st.sampled_from(GET_STATUS), GENERATE_NOTES, st.booleans())
+    def test_publication_detail_save_with_all_valid_fields(self, p, slug, platforms, sponsors, model_documentations, status, notes,
+                                                           flagged):
+        user = models(User).example()
```

Comment:
This should be in the given decorator as GENERATE_USER. See the comment below for detail.